pkgs/tools/networking/yggdrasil/default.nix

```diff
@@ -0,0 +1,36 @@
+{ pkgs, lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "yggdrasil";
+  version = "0.3.5";
+
+  src = fetchFromGitHub {
+    owner = "yggdrasil-network";
+    repo = "yggdrasil-go";
+    rev = "v${version}";
+    sha256 = "0cbj9hqrgn93jlybf3mfpffb68yyizxlvrsh1s5q21jv2lhhjcwj";
+  };
+
+  modSha256 = "1r942mqlq9lqxizxi33768jp4j706mwa2fliz0qbkh98r4ri2292";
+
+  subPackages = [ "." ];
+
+  buildPhase = ''
+    export PKGNAME="${pname}"
+    export PKGVER="${version}"
+    ./build
```

Comment:
the only good thing this custom build is getting you is the ldflags. Please add the ldflags manually and remove both of your custom phases.

Example:

https://github.com/NixOS/nixpkgs/blob/576ffb1f32267a022efc0976c4b95b85b4b5d16d/pkgs/development/tools/conftest/default.nix#L14-L17

Reply:
Ok, I tried it without the custom phases and with this:
```

 buildFlagsArray = ''
     -ldflags=
         -X github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.buildVersion=${version}
         -X github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.buildName=${pname}
  '';
```
which is what it looks to me like yggdrasil's build script is setting the ldflags to, see [here](https://github.com/yggdrasil-network/yggdrasil-go/blob/master/build).

Result:
```
can't load package: package github.com/yggdrasil-network/yggdrasil-go: unknown import path "github.com/yggdrasil-network/yggdrasil-go": cannot find module providing package github.com/yggdrasil-network/yggdrasil-go
builder for '/nix/store/4d1c005ypmly2vx45apn733jcs7f820s-yggdrasil-0.3.5.drv' failed with exit code 1
error: build of '/nix/store/4d1c005ypmly2vx45apn733jcs7f820s-yggdrasil-0.3.5.drv' failed
```

Reply:
Hi @kalbasit, the problem was that setting subPackages to `["."]` was not the right thing to do, and there is a bug in buildGoModule which prevents the derivation from building when you set subPackages to `["cmd/yggdrasil" "cmd/yggdrasilctl"]`.  I just submitted #63923 regarding the bug.